mlrun/utils/helpers.py

```diff
@@ -1015,7 +1015,16 @@ def create_function(pkg_func: str):
     splits = pkg_func.split(".")
     pkg_module = ".".join(splits[:-1])
     cb_fname = splits[-1]
+    pkg_module_exists_in_modules = pkg_module in sys.modules
     pkg_module = __import__(pkg_module, fromlist=[cb_fname])
+
+    # if the function appears in the modules list, we need to reload the code again because it may have changed
+    if pkg_module_exists_in_modules:
+        logger.warning(
+            "Reloading module - be aware that not all modules can be reloaded again"
+        )
+        reload(pkg_module)
```

Comment:
wrap in try/except to ensure that it wont fail. reload will recompilepython code and might fail so we want to avoid.
`also, worth doing it *only* when local=True`